distributed_shampoo/distributed_shampoo.py

```diff
@@ -1142,6 +1148,114 @@ def _per_group_step_impl(
             masked_blocked_search_directions=masked_blocked_search_directions
         )
 
+    @torch.no_grad()
+    def _per_group_step_experimental_lrs(
+        self,
+        state_lists: Dict[str, Any],
+        step: torch.Tensor,
+        neg_lrs: List[torch.Tensor],
+        beta1: float,
+        beta3: float,
+        weight_decay: float,
+        momentum_param: float,
+        dampening: float,
+        grafting_config_not_none: bool,
+        compute_root_inverse: bool,
+        use_decoupled_weight_decay: bool,
+        use_bias_correction: bool,
+        use_grafting_method: bool,
+        use_nesterov: bool,
+    ) -> None:
+        # Incorporate L2-regularization or (coupled) weight decay if enabled.
+        #   G <- G + lr * weight_decay * W
```

Comment:
Reminder to self that we have a typo here; shouldn't have `lr * weight_decay`, just `weight_decay`.

cc: @tsunghsienlee